app/models/plugins/thermometer.rb

```diff
@@ -28,4 +29,57 @@ def liquid_data
   def name
     self.class.name.demodulize
   end
+
+  def update_goal
+    self.goal = self.determine_next_goal
+    self.save
+  end
+
+  protected
+  def goal_should_update
+    current_total >= self.goal
```

Comment:
I might use some offset here (if the number is relatively close to the goal, rather than if the number is equal or larger than). I think this is a product question, though.